src/OpenSage.Game/Data/Rep/ReplayChunk.cs

```diff
@@ -54,7 +54,7 @@ internal static ReplayChunk Parse(BinaryReader reader)
                         break;
 
                     case OrderArgumentType.ObjectId:
-                        order.AddObjectIdArgument(reader.ReadUInt32());
+                        order.AddObjectIdArgument(new ObjectId(reader.ReadUInt32()));
```

Comment:
Is there a ReadObjectId method?

Reply:
Maybe there should be, but it's not easy to add as `BinaryReaderExtensions` live in `OpenSage.FileFormats` which doesn't (and cannot) reference `ObjectId`. We'd either need to add a separate set of binary reader extensions to this assembly, or to move `ObjectId` to `OpenSage.Core` and make `FileFormats` depend on that.

Reply:
I'd suggest adding a separate set of binary reader extensions to this assembly (or... keep it as it is, since it's not needed in many places).

Reply:
Looks like this is the only place for now where this exact pattern repeats (read uint32 + `new ObjectId`), so we probably don't need that extension for now.